algorithms/linfa-clustering/src/gaussian_mixture/algorithm.rs

```diff
@@ -211,6 +211,11 @@ impl<F: Float> GaussianMixtureModel<F> {
         self.means()
     }
 
+    pub fn predict_proba<D: Data<Elem = F>>(&self, observations: &ArrayBase<D, Ix2>) -> Array2<F> {
```

Comment:
Add a doc string, something like : "Returns the responsabilities as a (n_obs, n_clusters) array where each row gives the probabilities for the corresponding ith observation to belong to the jth cluster."